arches_component_lab/views/api/relatable_resources.py

```diff
@@ -34,44 +34,35 @@ def get(self, request, graph, node_alias):
             "name", "graphid"
         )
 
-        resources = ResourceInstance.objects.filter(graph_id__in=graphs).annotate(
-            order_field=Case(
-                When(resourceinstanceid__in=initial_values, then=Value(0)),
-                default=Value(1),
-            )
+        resources = ResourceInstance.objects.filter(graph_id__in=graphs).exclude(
+            resourceinstanceid__in=initial_values
         )
 
-        query_string = "descriptors__{}__name__icontains".format(language)
-
         if filter_term:
             resources = resources.filter(
-                Q(**{query_string: filter_term})
-                | Q(resourceinstanceid__in=initial_values)
+                 Q(**{"descriptors__{}__name__icontains".format(language): filter_term})
             )
 
-        paginator = Paginator(
-            resources.order_by(
-                "order_field", "descriptors__{}__name".format(get_language())
-            ),
-            items_per_page,
-        )
-        page_object = paginator.get_page(page_number)
+        offset = (int(page_number) - 1) * int(items_per_page)
+        limit = (int(offset)) + int(items_per_page)
```

Comment:
Instead of managing the offset and limit calculations ourselves, we can continue using the `Paginator` but just make the `count()` method a no-op.

I confirmed that this is enough to short-circuit the expensive count:
```py
        resources = resources.order_by("graph", "pk")
        resources.count = lambda self=None: 1_000_000_000
        paginator = Paginator(resources, items_per_page)
```

I'll open a ticket with Django about why the lambda has to take a self argument, seems unnecessary.

Then we can continue getting usability benefits like:
- errors if you provide an unordered queryset
- sensible behavior if the requested page is out of range

Reply:
> I'll open a ticket with Django about why the lambda has to take a self argument, seems unnecessary.

There's an older almost identical ticket [here](https://code.djangoproject.com/ticket/32784), so maybe we can just subclass the Paginator and override count instead of messing around with a lambda, should work fine.